2_base_and_sensor.ipynb

```diff
@@ -436,135 +812,259 @@
   },
   {
    "cell_type": "markdown",
-   "metadata": {},
+   "metadata": {
+    "lang": "en"
+   },
+   "source": [
+    "Sensor values are stored using Python \"data structure\".\n",
```

Comment:
If it only uses one Python data structure:

    "Sensor values are stored using a Python \"data structure\".\n",

If it uses multiple Python data structures:

    "Sensor values are stored using Python \"data structures\".\n",